machine/v1beta1/types_awsprovider.go

```diff
@@ -84,6 +84,13 @@ type AWSMachineProviderConfig struct {
 	// When omitted, no placement group is used when creating the EC2 instance.
 	// +optional
 	PlacementGroupName string `json:"placementGroupName,omitempty"`
+	// PlacementGroupPartition is the partition number within the placement group in which to launch the instance.
+	// This value is only valid if the placement group, referred in `PlacementGroupName`, was created with
+	// strategy set to partition.
```

Comment:
Does this field exist in the upstream CAPA? 

Godoc comments should start with json tag name not Go field name, so `placementGroupPartition` would be better.

Godoc should also explain the validations in place, so in this case, explain in the text that the value should be an integer value between 1 and 7.

Reply:
Yes, we have added the field in the upstream CAPA as well. Please see: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4883

Reply:
>Godoc comments should start with json tag name not Go field name, so `placementGroupPartition` would be better.

Okay. However, I observed that the existing Godoc comments for fields like `PlacementGroupName`, `MetadataServiceOptions`, etc., are also using the Go field name. The new field should be consistent with them as well.

Reply:
Prefer not to repeat mistakes of the past, that just adds to tech debt

Reply:
Okay, thanks. Updated.